### PR TITLE
specify backup version in backup operation

### PR DESCRIPTION
**Pull request by ryanfkeepers**
In order to test version-based changes at the operation level, we need to control what version is persisted in the backup.  This has some edge cases, for instance: setting the backup version doesn't actually force behavior to comply with that version's behavior.  It only writes that version to the backup model.  For that reason, new backup models default to the current version, and must be manually overwritten if a test needs a different variation.

---

#### Does this PR need a docs update or release note?

- [x] :no_entry: No

#### Type of change

- [x] :bug: Bugfix
- [x] :robot: Supportability/Tests

#### Issue(s)

* #2825

#### Test Plan

- [x] :zap: Unit test
- [x] :green_heart: E2E

### CODE REVIEW

File: src/internal/operations/backup.go [hunk 1]
```diff
@@ -36,9 +36,10 @@ type BackupOperation struct {
 
 	ResourceOwner common.IDNamer
 
-	Results   BackupResults      `json:"results"`
-	Selectors selectors.Selector `json:"selectors"`
-	Version   string             `json:"version"`
+	Results       BackupResults      `json:"results"`
+	Selectors     selectors.Selector `json:"selectors"`
+	Version       string             `json:"version"`
+	BackupVersion int                `json:"backupVersion"`
```
[the remark this answers is not in the document]
Comment (ashmrtn) [reply src/internal/operations/backup.go:42]: beyond that, I'm a little on the fence about having the backup operation keep the version that it's using. This assumes we can always determine the version up-front when the operation is created, before even examining what it is we're supposed to backup (i.e. Exchange, SharePoint) and how (if we have persisted feature flags on backups in the future, what feature flags were enabled on the base, etc).

Overall, I think it works well to get the test running, but I'm not sure if it will allow us to continue to expand Corso as a system without revisiting it. To pitch an alternative for getting testing to work, one thing that's possible is to add a wrapper mock around the call to get previous backups and modify things on the fly there. It does require more setup and such, but would allow us to delay the point where we actually need to say what version we want to place in the backup model

The above is more a discussion point rather than something that has to be done in this PR, but I think moving forward it's something we're going to have to figure out how to address

**Review by ashmrtn — APPROVED** [review]

File: src/internal/operations/backup.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (ryanfkeepers) [reply src/internal/operations/backup.go:42]: > Overall, I think it works well to get the test running, but I'm not sure if it will allow us to continue to expand Corso as a system without revisiting it.

Yep.  That's more or less all I'm after here.  It's a minefield already, and if it wasn't for testing requirements I wouldn't want it at all.

> add a wrapper mock around the call to get previous backups

That sounds like a healthy alternative overall.  Also, like you said, heavy.  I'll make a tech debt ticket for this, but am going to carry on with the current implementation.
